src/test/java/spoon/test/template/SubstitutionTest.java

```diff
@@ -47,4 +51,44 @@ public void testSubstitutionInsertAllFields() {
         @Override
         public void statement() { }
     }
+
+    @Test
+    public void testSubstitutionInsertAllNestedTypes() {
```

Comment:
It's already in the `SubstitutionTest` class, so no need to name all tests something with `Substitution`.

```suggestion
    public void testInsertAllNestedTypes() {
```